payjoin-cli/src/app/v2.rs

```diff
@@ -52,11 +53,12 @@ impl AppTrait for App {
             Some(send_session) => send_session,
             None => {
                 let psbt = self.create_original_psbt(&uri, fee_rate)?;
-                let mut req_ctx = SenderBuilder::new(psbt, uri.clone())
+                let mut persister = SenderPersister(self.db.clone());
+                let new_sender = SenderBuilder::new(psbt, uri.clone())
                     .build_recommended(fee_rate)
                     .with_context(|| "Failed to build payjoin request")?;
-                self.db.insert_send_session(&mut req_ctx, url)?;
-                req_ctx
+                let storage_token = new_sender.persist(&mut persister)?;
+                storage_token.load(persister)?
```

Comment:
This approach has the added benefit of ensuring the rountrip ser/deser application impl works before the type state machine can continue.
cc @DanGould @nothingmuch Lmk if this is what you had in mind.

Reply:
Hmm his wasn't exactly what I had in mind

- We should only have to define 1 generic persister::NoOpPersister. Because it does no operation, it should be able to handle whatever generic Key/Value/Token we give it. It probably doesn't need a hashmap even, it can just serialize it to bytes (Vec) if it can't be a newtype around a generic.
- The API should be symmetric, if possible.NewSender::persist and Sender::load should both be defined and viable given a Persister